### ISSUE TITLE
Keeping Chocolatey Package up to date

### ISSUE THREAD

**Issue by gep13** [keybase/client#10956]
I have recently created a Chocolatey Package for keybase, which allows the installation on a Windows machine using:

`choco install keybase`

This has been pushed, and is working well apart from one known issue.

My question is with regard to keeping that package up to date with releases that come out.  I would like to setup an automated task to check your website for the latest released version, and check that against what is published to Chocolatey, and if there is a new released version, push a new Chocolatey Package version.  Is there a page on your website that lists the currently released version of keybase client?  I had a quick look, but I wasn't able to find anything.  Thanks!

**Comment by strib** [comment]
@gep13 Thanks!  Your best bet is probably the github release tags, which roughly monitor our desktop releases for mac/windows.

**Comment by gep13** [comment]
@strib thanks for getting back to me...

I did think about using that, however, I noticed that they currently don't seem to be aligned.  For example, the latest GitHub Release is 1.0.44, and the latest installed version on my Windows machine is 1.0.45, which wouldn't really work in terms of verifying that the Chocolatey Package installs the correct latest version.

Is there anything else that I could use?

For example, is this download URL:

https://prerelease.keybase.io/keybase_setup_386.exe

Actually re-directing to a versioned location, or are you guys replacing the file at that location with each new release?

**Comment by strib** [comment]
Yeah the last few tags have been a bit off compared to the releases, but our goal is to keep them the same.  cc @cjb.

For Windows unfortunately I think we overwrite the release right now, but I'm not positive.  cc @zanderz.

**Comment by zanderz** [comment]
The source of truth for our updater is currently: http://s3.amazonaws.com/prerelease.keybase.io/update-windows-prod-v2.json
You can see that it points to https://prerelease.keybase.io/windows/Keybase_1.0.45-20180305114930%2B3a1c65a.386.exe
keybase_setup_386.exe is just renamed from the above semantic versioned format, which is unfortunate because it's hard to tell what version it is without at least downloading it, but it should generally always match what's in the update json metadata. Also there is no windows style version info in this update metadata - we could add it if it would be of use in situations like this.

**Comment by gep13** [comment]
>@zanderz said...
The source of truth for our updater is currently: http://s3.amazonaws.com/prerelease.keybase.io/update-windows-prod-v2.json

Ah, that sounds perfect!  Thank you very much!

>@zanderz said...
Also there is no windows style version info in this update metadata - we could add it if it would be of use in situations like this.

Can you confirm exactly what you mean by this?

**Comment by zanderz** [comment]
I mean that semantic version is of the form 1.0.45-20180316034714+bad7e6c, where windows versioning (also called "romantic versioning") is of the form 1.0.45.1 (4 numbers with dots in between). Our updater uses the former, but windows only understands the latter.

**Comment by zanderz** [comment]
I was wondering what the benefit of chocolatey is when Keybase has its own (incompatible I assume) update mechanism?

**Comment by gep13** [comment]
>@zanderz said...
I mean that semantic version is of the form 1.0.45-20180316034714+bad7e6c, where windows versioning (also called "romantic versioning") is of the form 1.0.45.1 (4 numbers with dots in between). Our updater uses the former, but windows only understands the latter.

Gotcha!  For my purposes, the four part version number will suffice.

>@zanderz said...
I was wondering what the benefit of chocolatey is when Keybase has its own (incompatible I assume) update mechanism?

Having a keybase package for Chocolatey comes with a number of advantages.

The first is ease of installation.  Assuming you already have Chocolatey installed, then installing it is as simple as typing:

`choco install keybase`

in the same way that I can on a Mac using brew, i.e.:

`brew cask install keybase`

Once you have a number of Chocolatey packages installed, keeping them all up to date is as simple as running:

`choco upgrade all`

In the case of keybase (as well as others, like Google Chrome) where it comes with it's own update mechanism, you can run the command:

`choco pin keybase`

So that Chocolatey no longer looks at keeping it updated, and defers to the built in installer.  In these cases, it is still useful to keep the Chocolatey package up to date with the latest version for the initial installation experience.
